### PR TITLE
Add support for dac ltc166x

### PR DESCRIPTION
**Pull request by marcusfolkesson**
Add support for LTC1665/LTC1660 8/10-bit Digital-to-Analog Converters

### CODE REVIEW

**Review by martinjaeger — CHANGES_REQUESTED** [review]
Thanks for the contribution. Looks very good, only got a small remark regarding coding style.

File: drivers/dac/dac_ltc166x.c [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+{
+	const struct ltc166x_config *config = dev->config;
+	uint16_t regval;
+
+	regval = FIELD_PREP(LTC166X_REG_MASK, addr);
+
+	if (config->resolution == 10)
+		regval |= FIELD_PREP(LTC166X_DATA10_MASK, data);
+	else
+		regval |= FIELD_PREP(LTC166X_DATA8_MASK, data);
```
[the remark this answers is not in the document]
Comment (marcusfolkesson) [reply drivers/dac/dac_ltc166x.c]: Thank you Martin,

I've updated the commit with your suggestion together with a couple of minor changes reported by checkpatch.

**Review by martinjaeger — DISMISSED** [review]

File: drivers/dac/dac_ltc166x.c [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+		LOG_ERR("unsupported channel %d", channel);
+		return -ENOTSUP;
+	}
+
+	if (value >= (1 << config->resolution)) {
+		LOG_ERR("Value %d out of range", value);
+		return -EINVAL;
+	}
+
+	ret = ltc166x_reg_write(dev, channel + 1, value);
```
Comment (benjaminbjornsson) [inline drivers/dac/dac_ltc166x.c:88]: Return `return ltc166x_reg_write(dev, channel + 1, value);`.

**Review by benjaminbjornsson — CHANGES_REQUESTED** [review]
Just a small comment, otherwise LGTM.

File: drivers/dac/dac_ltc166x.c (hunk 2 above)
Comment (marcusfolkesson) [reply drivers/dac/dac_ltc166x.c]: Good catch.

**Review by benjaminbjornsson — APPROVED** [review]

**Review by martinjaeger — APPROVED** [review]

**Comment by benjaminbjornsson** [comment]
@martinjaeger is there anything blocking this PR from being merged?

**Comment by martinjaeger** [comment]
> @martinjaeger is there anything blocking this PR from being merged?

No, ready to be merged.

**Pull request merged**